scylla-server-rust/src/controllers/car_command_controller.rs

```diff
@@ -0,0 +1,70 @@
+use std::sync::Arc;
+
+use axum::{extract::Path, Extension};
+use axum_extra::extract::Query;
+use protobuf::Message;
+use rumqttc::v5::AsyncClient;
+use serde::Deserialize;
+use tracing::{info, warn};
+
+use crate::{command_data::CommandData, error::ScyllaError};
+
+/// the prefix for the calypso topic, so topic of cmd is this plus the key appended on
+pub const CALYPSO_BIDIR_CMD_PREFIX: &str = "Calypso/Bidir/Command/";
+
+#[derive(Deserialize, Debug)]
+pub struct ConfigRequest {
+    pub data: Option<Vec<f32>>,
+}
+
+/// This controller recieves two peices of info: the key from the path, and the list of float values from the query params
+/// The key is used in accordance with the calypso specification: a command_data object is sent over siren to CALYPSO_BIDIR_CMD_PREFIX/<key>
+/// That command_data object is processed and matched via the key to a YAML specification for the encoding and sending of a CAN message.
+/// If the data is invalid, too short, etc. the default feild from the YAML is instead sent, as with when the car initially turns on before argos connects.
+/// TLDR: This triggers a command <key> with the values <data_query.0.data> for calypso to update the CAN packet sent to the bus.
```

Comment:
Why are you sending the default field if the data is invalid? Shouldn't it just reject it and send an error back to scylla?

Reply:
Also if you need to have a tl;dr on a comment, you should probably reword it a bit. I'd format it in some formal kind of way to describe params, errors, returns, etc. it makes it easier to read. I know java has a standard format for its comments, not sure if there's one for rust

Reply:
Ok will do.  Invalid is the wrong term I used.  If the data is the wrong number of points or is null, then the default is sent.  This is a way to "reset to default" the config setting.